lib/agent0/agent0/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -0,0 +1,532 @@
+        else:
+            self.config = config
+
+        # Define agent0 configs with this setup
+        # TODO currently getting the path based on this file's path
+        # This requires the entire monorepo to be check out, and will likely not work when
+        # installing agent0 by itself.
+        # This should get fixed when abis are exported in hypertypes.
+        full_path = os.path.realpath(__file__)
+        current_file_dir, _ = os.path.split(full_path)
+        abi_dir = os.path.join(current_file_dir, "..", "..", "..", "..", "..", "packages", "hyperdrive", "src", "abis")
+
+        self.eth_config = EthConfig(
+            artifacts_uri="not_used",
+            rpc_uri=chain.rpc_uri,
+            abi_dir=abi_dir,
+            preview_before_trade=self.config.preview_before_trade,
+        )
+
+        self.interface = HyperdriveReadWriteInterface(
+            self.eth_config,
+            hyperdrive_addresses,
+            web3=chain._web3,
+        )
+
+        self.chain = chain
+
+    def init_agent(
+        self,
+        private_key: str,
+        policy: Type[HyperdriveBasePolicy] | None = None,
+        policy_config: HyperdriveBasePolicy.Config | None = None,
+    ):
+        """Initializes an agent object given a private key.
+
+        .. note::
+            Due to the underlying bookkeeping, each agent object needs a unique private key.
+
+        Arguments
+        ---------
+        private_key: str
+            The private key of the associated account.
+        policy: HyperdrivePolicy, optional
+            An optional policy to attach to this agent.
+        policy_config: HyperdrivePolicy, optional
+            The configuration for the attached policy.
+
+        Returns
+        -------
+        HyperdriveAgent
+            The agent object for a user to execute trades with.
+        """
+        # If the underlying policy's rng isn't set, we use the one from interactive hyperdrive
+        if policy_config is not None and policy_config.rng is None and policy_config.rng_seed is None:
+            policy_config.rng = self.config.rng
+        out_agent = IHyperdriveAgent(
+            pool=self,
+            policy=policy,
+            policy_config=policy_config,
+            private_key=private_key,
+        )
+        return out_agent
+
+    def _init_agent(
+        self,
+        policy: Type[HyperdriveBasePolicy] | None,
+        policy_config: HyperdriveBasePolicy.Config | None,
+        private_key: str,
+    ):
+        # Setting the budget to 0 here, we'll update the wallet from the chain
+        agent = HyperdriveAgent(
+            Account().from_key(private_key),
+            initial_budget=FixedPoint(0),
+            policy=IHyperdrivePolicy(
+                IHyperdrivePolicy.Config(sub_policy=policy, sub_policy_config=policy_config, rng=self.config.rng)
+            ),
+        )
+
+        # TODO add the public address to the chain object to avoid multiple objects
```

Comment:
isn't this done? I thought that is what `chain._ensure_no_duplicate_addrs` is using?